src/Components/Artwork/Contact.tsx

```diff
@@ -19,42 +20,53 @@ export class Contact extends React.Component<ContactProps, null> {
 
   auctionLine() {
     const { artwork } = this.props
-    if (artwork.sale.is_live_open) {
+    const isLiveOpen = get(artwork, p => p.sale.is_live_open)
+    const isOpen = get(artwork, p => p.sale.is_open)
+    const isClosed = get(artwork, p => p.sale.is_closed)
+
+    if (isLiveOpen) {
       return (
         <TextLink href={artwork.href} underline>
           Enter Live Auction
         </TextLink>
       )
-    } else if (artwork.sale.is_open) {
-      const sa = artwork.sale_artwork
-      const bids = sa && sa.counts && sa.counts.bidder_positions
+    } else if (isOpen) {
+      const sa = get(artwork, p => p.sale_artwork)
+      const bids = get(sa, p => p.counts.bidder_positions)
+      const highestBidDisplay = get(sa, p => p.highest_bid.display)
+      const openingBidDisplay = get(sa, p => p.opening_bid.display)
```

Comment:
love that this is all type-safe

Reply:
Something I noticed when making the recent brick notifications was that there's an MP field that does this already. Basically, we want to display `highest_bid` if present, otherwise `opening_bid` if the auction is open.

Behold...the power of `current_bid`: https://github.com/artsy/metaphysics/blob/06c33e4cb03d8e41a1ad0e5b99db78b650081309/src/schema/sale_artwork.js#L116